### DISCUSSION TITLE
Add custom annotation to all generated resolver methods

### DISCUSSION THREAD

**Discussion by thisarattr** [kobylynskyi/graphql-java-codegen#783] [Q&A] [answered]
Hi all,
I wanted to add custom annotation to all generated resolver methods, So that  I can enforce authz on entry point to seperate data resources separately.

I achieved it using custom directive on schema which is added to all fields and use `directiveAnnotationsMapping` on  gradle plugin config to generate it. But still ppl can choose to add it or they can forget/miss it, which leads to hv the resource without any authz check. 
Thus im trying to add it to generation. Basically if I can  add some config saying I want this method annotation on all recover  interface methods, it will enforced.

Is there any way to me to achieve that?

Thanks in advance!

(last option im thinking is to hv some sort of linting to validate it and fail the build)

**Comment by kobylynskyi** [comment]
Hi @thisarattr, thanks for your request.
As I understand, you want to add a single annotation above all your Query/Mutation methods.
If that's true, then this is currently not supported. You can refer to the following class for details: https://github.com/kobylynskyi/graphql-java-codegen/blob/96c8c7f88265698c5e9705cd022bdbf61fd63076/src/main/java/com/kobylynskyi/graphql/codegen/mapper/GraphQLTypeMapper.java#L295
One thing we can do is to support regex in `customAnnotationsMapping` so that you can provide some wildcard in your config, e.g.:
```groovy
customAnnotationsMapping = ["Query.*":"com.yourAnnotation"]
```
Please let me know what you think.

**Comment by thisarattr** [comment]
Yes, that will work, so I get it to generate something like below. I will hv a look at the code 👍 
```
    @GqlSecured
    UserTO user(RootTO rootTO, String id, graphql.schema.DataFetchingEnvironment env) throws Exception;
```
Thanks!

**Comment by kobylynskyi** [comment, marked as answer]
@thisarattr the feature was released in 5.3.0.
Now you can supply a regex in `customAnnotationsMapping`. E.g.:
`customAnnotationsMapping = ["Query..*": ["com.your.organization.GqlSecured"], "Mutation..*": ["com.your.organization.GqlSecured"]]`
or even:
`customAnnotationsMapping = [".*": ["com.your.organization.GqlSecured"]]`
Please try it out and let me know how it works.

**Reply by s-af** [reply]
@kobylynskyi, Is it possible to use regex for customAnnotationsMapping in maven plugin?
